object/interp.go

```diff
@@ -2,22 +2,40 @@ package object
 
 import (
 	"errors"
+	"fmt"
+	"strings"
+
+	"grol.io/grol/lexer"
 )
 
+type ExtensionMap map[string]Extension
+
 var (
-	extraFunctions   map[string]Extension
+	extraFunctions   ExtensionMap
 	extraIdentifiers map[string]Object
 	initDone         bool
 )
 
 // Init resets the table of extended functions to empty.
 // Optional, will be called on demand the first time through CreateFunction.
 func Init() {
-	extraFunctions = make(map[string]Extension)
+	extraFunctions = make(ExtensionMap)
 	extraIdentifiers = make(map[string]Object)
 	initDone = true
 }
 
+func ValidIdentifier(name string) bool {
+	if name == "" {
+		return false
+	}
+	for _, b := range []byte(name) {
+		if !lexer.IsAlphaNum(b) {
+			return false
+		}
+	}
```

Comment:
Usually it's better to loop on runes from a string

Unicode characters such as accent, and emoji are badly checked otherwise

Then Unicode package can be used to check the type

Reply:
it's on purpose (for now at least) only bytes (ascii) is allowed outside of strings

unicode does work fine inside strings (as demoed in many examples)